zerver/actions/create_user.py

```diff
@@ -343,7 +343,7 @@ def stream_name(stream: Optional[Stream]) -> Optional[str]:
         default_sending_stream=default_sending_stream_name,
         default_events_register_stream=default_events_register_stream_name,
         default_all_public_streams=user_profile.default_all_public_streams,
-        avatar_url=avatar_url(user_profile),
+        avatar_url=avatar_url(user_profile, client_gravatar=True),
```

Comment:
We would need to set `client_gravatar` as `False` here, since we would not know whether client can compute the gravatar or not. `notify_created_user` has similar implementation.

Reply:
@sahil839 making `client_gravatar` always equal to `False` here will cause various backend tests related to bots to fail. The reason for this is that many tests are calling the function implemented [here](https://github.com/zulip/zulip/blob/main/zerver/tests/test_events.py#L249). As this function matches the state received from server and states after applying events. As in states received  from event `avatar_url` is always sent and in state received from server `avatar_url` is `None` states does not match and hence tests are failed.

Reply:
We cannot assume that client can compute url for all clients by passing `client_gravatar` as `True` just for the sake of tests. Otherwise, the clients who cannot compute the url will also not recieve URL and would not be able to render the avatar correctly. 

Instead we should assume that client cannot compute gravatar and send the url to all clients. For the tests, we would need to add code to `events.py` which sets the `avatar_url` to `None` if client can compute it. See the code for `realm_user/add` type event in `lib/events.py`.

Reply:
Thanks! I was able to figure out how to fix the failing test cases.